homeassistant/components/wsdot/__init__.py

```diff
@@ -1 +1,18 @@
 """The wsdot component."""
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up wsdot as config entry."""
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    return True
```

Comment:
Ideally we do a check if the API key we have works, otherwise we would create unavailable entities that would keep polling individually.

You can then store the client you created in `entry.runtime_data`

Reply:
I can check that the key is valid here, but there is only one apikey necessary to access any of this department's 16 separate services. And so there is not just one entry to store in runtime_data. I could store the travel_time specific client in runtime_data and then whenever I add a new API domain, it could also get added to runtime_data. It just felt wasteful to pre-create all of these when the user probably won't want all of them.